### PR TITLE
Feature/uot 125706 topics training api

### PR DESCRIPTION
**Pull request by dakotahavel**

### CODE REVIEW

File: gamechangerml/__init__.py [hunk 1]
```diff
@@ -2,9 +2,9 @@
 
 # abs path to this package
 PACKAGE_PATH: str = os.path.dirname(os.path.abspath(__file__))
-REPO_PATH: str = os.path.abspath(os.path.join(PACKAGE_PATH, '..'))
-DATA_PATH: str = os.path.join(PACKAGE_PATH, 'data')
-MODEL_PATH: str = os.path.join(PACKAGE_PATH, 'models')
-CORPUS_PATH: str = os.path.join(PACKAGE_PATH, 'corpus')
-NLTK_DATA_PATH: str = os.path.join(DATA_PATH, 'nltk_data')
-AGENCY_DATA_PATH: str = os.path.join(DATA_PATH, 'agencies')
+REPO_PATH: str = os.path.abspath(os.path.join(PACKAGE_PATH, ".."))
```
Comment (dakotahavel) [inline gamechangerml/__init__.py:5]: Sorry the formatter changed ' to "

File: gamechangerml/api/fastapi/routers/controls.py [hunk 2]
```diff
@@ -332,6 +375,140 @@ async def download_corpus(corpus_dict: dict, response: Response):
     return await get_process_status()
 
 
+# Create a mapping between the training methods and input from the api
+# Methods for all the different models we can train
+# Defined outside the function so they arent recreated each time its called
+
```
Comment (dakotahavel) [inline gamechangerml/api/fastapi/routers/controls.py:381]: Moved the training functions outside the api function to keep them from being recreated every time

File: gamechangerml/api/fastapi/routers/controls.py [hunk 3]
```diff
@@ -341,127 +518,15 @@ async def train_model(model_dict: dict, response: Response):
...
 
-        training_thread = MlThread(training_method)
+        build_type = model_dict.get("build_type")
+        training_method = training_switch.get(build_type)
+
+        if not training_method:
+            raise Exception(f"No training method mapped for build type {build_type}")
+
+        # Set the training method to be loaded onto the thread
+        training_thread = MlThread(training_method, args={"model_dict": model_dict})
```
Comment (dakotahavel) [inline gamechangerml/api/fastapi/routers/controls.py:529]: With functions moved outside the scope, the model dict arg has to be passed to the thread

File: gamechangerml/api/utils/status_updater.py [hunk 4]
```diff
@@ -0,0 +1,47 @@
+import logging
+import typing as t
+from gamechangerml.api.utils import processmanager
+
+logger = logging.getLogger("gamechanger")
+
+
+class StatusUpdater:
```
Comment (dakotahavel) [inline gamechangerml/api/utils/status_updater.py:8]: Created a helper class to iterate through process states, used in topic_modeling.train_from_files

File: gamechangerml/api/fastapi/routers/search.py [hunk 5]
```diff
@@ -61,10 +61,12 @@ async def textExtract_infer(body: dict, extractType: str, response: Response) ->
             logger.info(topics)
             results["extracted"] = topics
         elif extractType == "summary":
-            summary = GensimSumm(
-                query_text, long_doc=False, word_count=30
-            ).make_summary()
-            results["extracted"] = summary
+            # gensim upgrade breaks GensimSumm class
```
Comment (dakotahavel) [inline gamechangerml/api/fastapi/routers/search.py:64]: gensim upgrade breaks GensimSumm class, but this extractType is unused I'm told

**Review by rha930 — APPROVED** [review]

**Pull request merged**